src/common/SteelSheets.cpp

```diff
@@ -0,0 +1,137 @@
+#include "SteelSheets.hpp"
+#include "cmath_ext.h"
+#include <float.h>
+#include <string.h>
+#include <cmath>
+#include <algorithm>
+
+uint32_t SteelSheets::NextSheet() {
+    uint8_t index = ActiveSheetIndex();
+
+    for (size_t i = 1; i < MAX_SHEETS; ++i) {
+        if (IsSheetCalibrated((index + i) % MAX_SHEETS)) {
+            SelectSheet((index + i) % MAX_SHEETS);
+            return (index + i) % MAX_SHEETS;
+        }
+    }
+    // no calibrated sheet, set active sheet to index 0
+    SelectSheet(0);
+    return 0;
+}
+bool SteelSheets::IsSheetCalibrated(uint32_t index) {
+    Sheet sheet = getSheet(index);
+    return !nearlyEqual(sheet.z_offset, FLT_MAX, 0.001f);
```

Comment:
It is fragile and hard to read use FLT_MAX directly here.
Consider naming that constant. E.g. in eeprom.h:
``` c++
#endif //__cplusplus
inline constexpr float eeprom_z_offset_uncalibrated = FLT_MAX;
```
in eeprom_v9.hpp:
``` c++
    { "Smooth2", eeprom_z_offset_uncalibrated },
    { "Textur1", eeprom_z_offset_uncalibrated },
    { "Textur2", eeprom_z_offset_uncalibrated },
    { "Custom1", eeprom_z_offset_uncalibrated },
    { "Custom2", eeprom_z_offset_uncalibrated },
    { "Custom3", eeprom_z_offset_uncalibrated },
    { "Custom4", eeprom_z_offset_uncalibrated },
```
And here
```
return !nearlyEqual(sheet.z_offset, eeprom_z_offset_uncalibrated , 0.001f);
```